pallets/authors-manager/src/lib.rs

```diff
@@ -365,18 +378,9 @@ impl<T: Config> Pallet<T> {
             .map_err(|_| Error::<T>::InvalidPublicKey)?;
         let author_id_bytes =
             <T as pallet::Config>::AccountToBytesConvert::into_bytes(author_account_id);
-        let function_name = b"addAuthor";
 
-        let params = vec![
-            (b"bytes".to_vec(), decompressed_eth_public_key.to_fixed_bytes().to_vec()),
-            (b"bytes32".to_vec(), author_id_bytes.to_vec()),
-        ];
-        let tx_id = <T as pallet::Config>::BridgeInterface::publish(
-            function_name,
-            &params,
-            PALLET_ID.to_vec(),
-        )
-        .map_err(|e| DispatchError::Other(e.into()))?;
+        let tx_id =
+            Self::publish_to_bridge(&decompressed_eth_public_key, &author_id_bytes, b"add_author")?;
```

Comment:
Unless I am missing something, the order of the params between `addAuthor` and `removeAuthor` are not the same which would break the encoding and invalidate signatures on Tier1. If we want to centralise the function, maybe we should pass in the `params` vector instead of the individual fields.

And perhaps we should have tests to ensure we don't accidentally break this logic as its the second time we are finding issues.